cli/bin/grain.js

```diff
@@ -211,6 +211,7 @@ program
 program
   .command("doc <file>")
   .description("generate documentation for a grain file")
+  .option("-o <filename>", "output filename")
```

Comment:
Oh, actually, looking at the format command, I think we need:
```suggestion
  .forwardOption("-o <filename>", "output filename")
```

Reply:
I tried it at first, but it fails with "option `-o' cannot be repeated".

Reply:
Hmm, @ospencer wrote that logic, so we'll need to ping him about it.

Reply:
`.forwardOption` like this seems to conflict with the `-o` from the main `grain` command.

Reply:
@cician correct, but those options are supposed to propagate down to the individual commands and show in the help, so this needs to actually get fixed and not worked around.

Reply:
I'm not sure if I understand correctly, but I think I found the root of the problem.

In exec.js the `execGraindoc` function merges the options from the main `grain` "program" and the `doc` "program" in order to inherit compiler flags. This results in the `flags` array having the `-o` twice and Cmdliner doesn't accept it. My version with `.option` only worked because this merging code skips non "forward" options, leaving just the inherited one.

I apologize if I'm off-track, but is this even necessary for the doc command to receive compilation flags? At a quick glace I don't see it actually using them, but I'm probably missing something. If it is necessary, do we want to just make an exception for the `-o` param or dynamically deduplicate them?

It was probably better if I just filed an issue. Now I'm in a new rabbit hole. 🕳

Reply:
Yes, since both the graindoc and grainformat tools inherit parts of the compiler, they implicitly need to accept arguments from the compiler. The option argument logic is really messy/confusing and we are hoping to clean it up for silo.

Reply:
I made changes to use `.forwardOption`. For now with an explicit exception for `-o`. I can change it if you think there's a better way.